contrib/native/client/patches/zookeeper-3.4.6-x64.patch

```diff
@@ -1,163 +0,0 @@
-From 64697ddd8a90f29d1693658f04e975e435e3c869 Mon Sep 17 00:00:00 2001
```

Comment:
I don't know.  Patch files are supposed to be excluded by the .gitignore.  What happened was I reapplied Drill's `.gitignore` and it deleted that patch file.   I would assume this file should not be included.

Reply:
It's very old:
```
commit d944918ff8a1f76d9b4c400512a130bff7fb2c5f
Author: Parth Chandra <pchandra@maprtech.com>
Date:   Wed Jun 11 18:10:34 2014 -0700

    DRILL-1021: Windows build
```
@vdiravka any thoughts about this patch file?

Reply:
If this PR really wants to be the one to delete this file from version control then I think it should include some comment  in the description like "this patch was added for X and is no longer relevant". I can see that it's a C code patch to ZooKeeper with a title of `[PATCH] Allow zookeeper to build in x64` which makes me 99% sure it's completely obsolete. @vdiravka, @vvysotskyi any advice for us here?

Reply:
There is [readme.win.txt](https://github.com/apache/drill/blob/master/contrib/native/client/readme.win.txt#L135) about it:
> The 3.4.6 release of Zookeeper does not build correctly on 64 bit windows. To fix that for the 64 bit build, apply patch zookeeper-3.4.6-x64.patch

Reply:
@vdiravka Forgive my lack of knowledge here, but the current `.gitignore` excludes patch files.  Is there a way to ignore all patch files, but leave that one?  How would you suggest we fix this?

@jnturton Aside from this patch file, are we good to go, pending CI?

Reply:
I think the .gitignore is working properly already because all of our previous commits have left zookeeper-3.4.6-x64.patch in peace. How about we do it this way? If this PR wants to delete the patch then it should simultaneously update readme.win.txt loosely saying "ZooKeeper 3.4.6 is ancient, ZooKeeper 3.5.7 is current and does not need any patch to build on Windows so the patch remains in the Git history only". Alternatively, we revert the deletion here and carry the cruft a bit longer.

+1 one from me otherwise.

Reply:
Sounds good.  Updating README.

Reply:
Done!

Reply:
@cgivre In any case, if the issue persists, there is a [way](https://stackoverflow.com/questions/987142/make-gitignore-ignore-everything-except-a-few-files) how to exclude only one file from the pattern:
`!contrib/native/client/patches/zookeeper-3.4.6-x64.patch`